src/lib/instantiate_bridge.ts

```diff
@@ -1,34 +1,41 @@
-import { BridgeHelper, networks, Bridge } from 'arb-ts';
 import { providers, Wallet, VoidSigner } from 'ethers';
 import args from './getClargs';
 import dotenv from 'dotenv';
+import { getL1Network, getL2Network, MultiCaller, TokenBridger } from "arb-ts"
 dotenv.config();
 
-const networkID = args.networkID?.toString() || '1';
+const networkID = args.l2NetworkID?.toString() || '42161';
 
-console.log('Using L1 networkID:', networkID);
+console.log('Using L2 networkID:', networkID);
 
+export const getNetworkConfig = async () => {
+  const l2Network = await getL2Network(networkID)
+  const l1Network = await getL1Network(l2Network.partnerChainID)
 
-export const instantiateBridge = async (
-) => {
-  const network = networks[networkID];
-  if (!network) {
-    throw new Error(`Unrecognized network ID: ${networkID}`);
-  }
-
-  const l1Network = network.isArbitrum
-    ? networks[network.partnerChainID]
-    : network;
-  const l2Network = networks[l1Network.partnerChainID];
-
-  const ethProvider = new providers.JsonRpcProvider(l1Network.rpcURL);
   const arbProvider = new providers.JsonRpcProvider(l2Network.rpcURL);
+  const ethProvider = new providers.JsonRpcProvider(l1Network.rpcURL)
 
   // random address for void signer:
   const l1Signer = new VoidSigner("0xAddA0B73Fe69a6E3e7c1072Bb9523105753e08f8", ethProvider) 
```

Comment:
Looks like we can do away with these signers altogether now

Reply:
:+1: